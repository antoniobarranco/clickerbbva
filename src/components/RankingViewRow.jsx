import React from 'react'
import PropTypes from 'prop-types';

import { formatNumericValue } from '../helpers/GameLogic';

const RankingViewRow = ({user, score}) => {
  return (
      <li className="list-group-item d-flex justify-content-between align-items-center">
        {user}
      <span className="badge bg-secondary rounded-pill">{formatNumericValue(score)}</span>
    </li>
  )
}

RankingViewRow.propTypes = {
  user: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
};

export default RankingViewRow