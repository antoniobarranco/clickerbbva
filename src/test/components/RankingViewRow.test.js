import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import RankingViewRow from "../../components/RankingViewRow";

describe('Tests on RankingViewRow component', () => {

    test('Test RankingViewRow with Snapshot test for UI changes', () => {

        const user = "username";
        const score = 1024;

        const tree = renderer.create(<RankingViewRow user={user}  score={score} />).toJSON();

        expect(tree).toMatchSnapshot();

    })

    test('Test RankingViewRow component shows propertly with their props', () => {

        const user = "username";
        const score = 1024;

        render(<RankingViewRow user={user}  score={score} />);

        expect(screen.getByTestId('RankingRow')).toHaveTextContent(user);
        expect(screen.getByTestId('RankingRow')).toHaveTextContent('1.02K Hits');

    })

})