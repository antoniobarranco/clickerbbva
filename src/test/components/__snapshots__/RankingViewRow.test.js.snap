// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on RankingViewRow component Test RankingViewRow with Snapshot test for UI changes 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
  data-testid="RankingRow"
>
  username
  <span
    className="badge bg-secondary rounded-pill"
  >
    1.02K Hits
  </span>
</li>
`;
